### ISSUE TITLE
Building with Carthage throws many warnings

### ISSUE THREAD

**Issue by jdanthinne** [OneSignal/OneSignal-iOS-SDK#206]
The latest releases throws warnings when using Carthage…

```
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:294:17: warning: class property 'lastMessageIdFromAction' requires method 'lastMessageIdFromAction' to be defined - use @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:54:25: warning: incompatible pointer to integer conversion assigning to 'UIBackgroundTaskIdentifier' (aka 'unsigned int') from 'void *' [-Wint-conversion]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:294:17: warning: class property 'lastMessageIdFromAction' requires method 'lastMessageIdFromAction' to be defined - use @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:54:25: warning: incompatible pointer to integer conversion assigning to 'UIBackgroundTaskIdentifier' (aka 'unsigned long') from 'void *' [-Wint-conversion]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:294:17: warning: class property 'lastMessageIdFromAction' requires method 'lastMessageIdFromAction' to be defined - use @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:54:25: warning: incompatible pointer to integer conversion assigning to 'UIBackgroundTaskIdentifier' (aka 'unsigned int') from 'void *' [-Wint-conversion]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:294:17: warning: class property 'lastMessageIdFromAction' requires method 'lastMessageIdFromAction' to be defined - use @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:54:25: warning: incompatible pointer to integer conversion assigning to 'UIBackgroundTaskIdentifier' (aka 'unsigned int') from 'void *' [-Wint-conversion]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:294:17: warning: class property 'lastMessageIdFromAction' requires method 'lastMessageIdFromAction' to be defined - use @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:54:25: warning: incompatible pointer to integer conversion assigning to 'UIBackgroundTaskIdentifier' (aka 'unsigned long') from 'void *' [-Wint-conversion]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:294:17: warning: class property 'lastMessageIdFromAction' requires method 'lastMessageIdFromAction' to be defined - use @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:54:25: warning: incompatible pointer to integer conversion assigning to 'UIBackgroundTaskIdentifier' (aka 'unsigned int') from 'void *' [-Wint-conversion]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:294:17: warning: class property 'lastMessageIdFromAction' requires method 'lastMessageIdFromAction' to be defined - use @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:54:25: warning: incompatible pointer to integer conversion assigning to 'UIBackgroundTaskIdentifier' (aka 'unsigned long') from 'void *' [-Wint-conversion]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:294:17: warning: class property 'lastMessageIdFromAction' requires method 'lastMessageIdFromAction' to be defined - use @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:54:25: warning: incompatible pointer to integer conversion assigning to 'UIBackgroundTaskIdentifier' (aka 'unsigned int') from 'void *' [-Wint-conversion]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:294:17: warning: class property 'lastMessageIdFromAction' requires method 'lastMessageIdFromAction' to be defined - use @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:54:25: warning: incompatible pointer to integer conversion assigning to 'UIBackgroundTaskIdentifier' (aka 'unsigned int') from 'void *' [-Wint-conversion]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:294:17: warning: class property 'lastMessageIdFromAction' requires method 'lastMessageIdFromAction' to be defined - use @dynamic or provide a method implementation in this class implementation [-Wobjc-property-implementation]
Carthage/Checkouts/OneSignal-iOS-SDK/iOS_SDK/OneSignalSDK/Source/OneSignalTracker.m:54:25: warning: incompatible pointer to integer conversion assigning to 'UIBackgroundTaskIdentifier' (aka 'unsigned long') from 'void *' [-Wint-conversion]
```

**Comment by jkasten2** [comment]
@jdanthinne Thanks for the report. You can safely ignore these warnings.  We will get them cleaned up in a future release.

**Comment by jkasten2** [comment]
@jdanthinne This is now fixed in release 2.4.2.
